test/e2e/non_typed_array_fields_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestNonTypedArrayFields(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	yaml := `
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: simple-svc
+  labels:
+    app: simple-svc
+spec:
+  ports:
+  - port: 80
+    name: test
+  clusterIP: None
+  selector:
+    app: simple-svc
+---
+apiVersion: apps/v1
+kind: StatefulSet
```

Comment:
nit: I think we changed it to a StatefulSet to address the example in the original issue,

However, I think a Deployment pod template could have both `envFrom: null` and `initContainers:`.
Thought I would point that out since a StatefulSet does more than a Deployment and that is not something we require for this case.

Reply:
Yeah, I was thinking of just keeping the StatefulSet, right now I think the Service and the ConfigMap don't mean anything in the test. We can keep the previous example of Deployment and ConfigMap if we want, and add the initContainers along with envFrom 👍🏻